### ISSUE TITLE
Define aliases for types and members using :=

### ISSUE THREAD

**Issue by VBAndCs** [dotnet/vblang#442]
Define aliases for types and members using :=

Today, `Using` allow some limited form of aliasing. I suggest to use `:=` instead to allow class-level and local aliasing. For example, the following code is not allowed:
```VB.NET:
  Dim Report = AddressOf Console.WriteLine
  Report("test")
```

Because `Console.WriteLine` has many overloads, the `Report` var is inferred to be of type Object! Generally, type inference doesn't work with AddressOf!
We have to define a delegate for only one overload:
```VB.NET:
  Dim Report As Action(Of String) = AddressOf Console.WriteLine
  Report("test")
```

My suggestion. is to use an alias instead:
```VB.NET:
  Dim Report := Console.WriteLine
  Report("test")
  Dim n = 1
  Report("test {0}", n)
  Dim GetFnwe := IO.Path.GetFileNameWithoutExtension
  Report(GetFnwe("C:\1.txt"))
  Dim Dic := Dictionary(Of Integer, List(Of String))
  Dim D As New Dic()
  D.Add(1, New List(Of String))
  Dim student := (Id as Integer, Name As String, Grade As Integer)
  Dim Adam As Student
  Adam.Name = "Adam"
```

This can be combined with #438  to get:
```VB.NET:
With n = 1,
     Report := Console.WriteLine, 
     GetFnwe := IO.Path.GetFileNameWithoutExtension, 
     Dic := Dictionary(Of Integer, List(Of String)),
     D As New Dic(),
     student := (Id as Integer, Name As String, Grade As Integer),
     Adam As Student

  Report("test")
  Report("test {0}", n)
  Report(GetFnwe("C:\1.txt"))
  D.Add(1, New List(Of String))
  Adam.Name = "Adam"
End With
```

And as I argued there, `=` should be used to assign scoped variables, while `:=` should be used for aliasing.
And of course defining `Dic` and `Student` at the class level will be better, to use them all over the class scope. This is just a quick example for demonstration .

**Comment by bandleader** [comment]
VB currently allows `Imports` to be used at the top of a file for type aliasing:
```vb
Imports StringList = List(Of String)```
I would extend _that_ to do what you are requsesting:
```vb
Imports Log = Console.WriteLine```

One caveat is that currently `Imports` statements cannot reference each other -- I would think that this should be lifted if we're going to do this -- otherwise you'll have to use the fully qualified namespace of `Console`.

**Comment by VBAndCs** [comment]
Yes. I meant `Imports` but I wrote the C# keyword `using` instead! Sorry.
`Imports` doesn't work for members nor tuples. So, `Imports Report = System.Console.WriteLine` is not accepted.
If `imports` can be modified to do that, I am OK with it. And perhaps it should be allowed inside local methods too.

**Comment by rskar-git** [comment]
One can put this at the top:

    Imports System.Console

And then one can do this in the code:

    WriteLine("Howdy")

And there you go, no need to say `Console` every time. :)

**Comment by VBAndCs** [comment]
OK. But we need aliasing for tuples too.
My goal here is to control the alias scope, as we do with local variables. If imports can be used locally, I am OK with that, as I don't care about the syntax.

**Comment by DualBrain** [comment]
I can't put my finger on the reason as to why... but something about this alias thing just doesn't seem like the right answer.  I think maybe I'm foreseeing many, many, many problems down the road especially with having to maintain software where this is "over utilized".  It just doesn't feel right.

**Comment by pricerc** [comment]
@DualBrain 
it's a bit like #168 that you also commented on recently.

**Comment by DualBrain** [comment]
@pricerc Agreed.  Although I get that we are trying to resolve some sort of underlying (universal) problem here... these just don't feel like the right answers.  Truly, I don't think there is an answer to this "problem".  If a solution to one problem opens us up to ten more... then it's not a solution.  What would happen if someone used something like this to "alias" nearly everything.  Imagine for a moment that you then had to jump in and take over / maintain this project.  Worse yet... imagine that you were the person that did this... you then move on to other projects... you then return a few years later to the project.  Would you remember what Pr, D, or Dic stood for.  These shorthand names are a bad idea for variables... now we want them to for "keywords"?  As I said, this doesn't "feel right".  Furthermore, using the example of Console.WriteLine… aliasing it isn't really a solution as you can see from the discussion on #103.  It's a bit more complicated than simply giving it a different name.  Using #103 as an example, ? is being proposed as an "alias" for Console.WriteN.  However, it's not simply an alias... there are other things to take into account.   Possibly what kind of project?  Should it have other things about it that further differentiate it from simply being a basic alternate to Console.WriteN.  If it is simply a wrapper, then one could just simply create a function called P and be done with it.  What I'd like to see with ? (or PRINT) is to actually look at the history of PRINT as a whole and possible reintroduce it into VB; where it would have the ability to work regardless of the project type.  This is way beyond aliasing... however, if done right it also adds value as a significant alternative to what is currently available and, furthermore, "feels right".

**Comment by pricerc** [comment]
@DualBrain . Agreed. A few of the proposals I've seen look to me like a solution looking for a problem. I think this is one of those.

As I've said elsewhere, one of the big things I like about VB, and why I often use it over C# when given the choice, is that it is legible by people who are not necessarily programmers. Part of this is that (I think) its existing conventions encourage that. I don't think it needs any more 'abbreviating' of any of the core language syntax, which this is advocating, as it would undermine long-term maintainability of a code base.

These kinds of things just lead to obfuscated code; there are better languages than VB for that.
